test/memoryPoolAPI.cpp

```diff
@@ -282,3 +285,37 @@ TEST_F(test, getLastFailedMemoryProvider) {
                   umfMemoryProviderGetName(umfGetLastFailedMemoryProvider())),
               "provider2");
 }
+
+struct poolHandleCheck
+    : umf_test::test,
+      ::testing::WithParamInterface<
+          std::function<std::variant<void *, umf_result_t, size_t>(void)>> {};
+
+TEST_P(poolHandleCheck, poolHandleCheckAll) {
```

Comment:
Could you add few comments on how this works? For someone with a "moderate" level of C++ knowledge? :) You are testing here the arguments of the functions or the result?

Reply:
Done. I've added some comments.

A longer explanation:
The idea here is to just verify if calling each of the functions listed below (umfPoolMalloc, umfPoolFree, ...) with NULL handle returns an appropriate error (NULL in case of a function returning void*, INVALID_ARGUMENT for functions returning status). But there is a problem: since those functions have different return types and different parameters we cannot just instantiate this TEST_P with function pointers directly. We need to have a single function pointer type. To do that I unify the return type using std::variant and `withGeneratedArgs` to unify parameters (which basically just sets each argument to NULL/zero). Now, we have a single function pointer signature (`std::variant<void *, umf_result_t, size_t>(void)`) and we can bind it to each umf function we are interested in.

I actually want to use a similar approach for testing the parameter validation in general where we could just define what combination or arguments we want to test.